src/transformers/generation/tf_utils.py

```diff
@@ -2267,12 +2267,14 @@ def unflatten_beam_dim(tensor, num_beams, batch_axis=0):
         decoder_hidden_states = [] if (return_dict_in_generate and output_hidden_states) else None
 
         # 3. init tensors to use for "xla-compileable" generate function
-        batch_size, num_beams, cur_len = shape_list(input_ids)
+        batch_size, num_beams, decoder_prompt_len = shape_list(input_ids)
```

Comment:
I see where you're going with this change, but I'm going to ask for a renaming: `cur_len` is used across the codebase as the length of `input_ids`, i.e. including the prompt.

For instance, we can hold a copy (`decoder_prompt_length = cur_len` here), and use it where needed.

Reply:
Yes, I also had this concern when doing the fix, I have updated the fix so that `cur_len` now represents the length of the entire sequence including the decoder prompt.